### PR TITLE
Implement cache for DelegateParameter to mirror cache of the source parameter

### LINKED ISSUE #1782: Bug in DelegateParameter: get_latest and set_cache are not forwarded to source parameter
See the following tests of `DelegateParameter` that are marked as `xfail`: https://github.com/QCoDeS/Qcodes/pull/1757/commits/b04266b94ff362d0829cdbd8d3215ca557009e2d ([on master around here](https://github.com/QCoDeS/Qcodes/blob/master/qcodes/tests/test_parameter.py#L1743-L1844))

Expectation: `DelegateParameter`s `get_latest` and `set_cache` should in principle simply forward the call to `get_latest` and `set_cache` of the source parameter.

### PR DESCRIPTION
**Pull request by Dominik-Vogel**
This PR implements cache of `DelegateParameter` to exactly mirror the cache of the source parameter. Depends on #1832.

For now, some dirty stuff with respect to accessing `_value` of cache is needed, but this wouldn't be necessary when `snapshot_base` improvements will be worked on in #1833.

Fixes #1782.

### CODE REVIEW

**Comment by astafan8** [comment]
@Dominik-Vogel could you please rebase this PR on top of https://github.com/QCoDeS/Qcodes/pull/1832 ?

**Comment by Dominik-Vogel** [comment]
done

**Comment by Dominik-Vogel** [comment]
(Rebased onto wrong branch, really sorry, I lost the overview over all the different cache PRs)
Now it is updated and sound, @astafan8 .

File: qcodes/instrument/parameter.py [hunk 1]
```diff
@@ -1287,6 +1265,58 @@ class DelegateParameter(Parameter):
     without changing those in the source parameter
     """
 
+    class _DelegateCache():
+        def __init__(self,
+                     source: _BaseParameter,
+                     parameter: _BaseParameter):
+            self._source = source
+            self._parameter = parameter
+
```
Comment (astafan8) [inline qcodes/instrument/parameter.py]: until the snapshot_base improvement #1833 , delegate cache has to also have `_value` forwarded to `source.cache._value`, sorry :(

Comment (Dominik-Vogel) [reply qcodes/instrument/parameter.py]: :-( Okay

**Comment by astafan8** [comment]
@Dominik-Vogel I merged the master with the `_Cache` into this branch (didn't want to bother with rebases), and i think it looks good. Let's have another last look and merge (if CI allows :) )

**Pull request merged**